### PR TITLE
Fix lti_section_owner_changed metric

### PR DESCRIPTION
**Pull request by carl-codeorg**
The original ticket for this was to add the missing `lms_name` property to the event metadata. While testing this change, I also noticed that events were being published even when the section owner did not change. This was happening because the IDs were strings when pulled from the params. Rails automatically casts the strings to ints, so the controller was working, but when we compared the new and old owner IDs to determine whether or not to publish the `lti_section_owner_changed`, they never matched, because the real ID is never a string.

This PR:
- Adds the missing `lms_name` property
- Stringifies the JSON on the client side, and parses it in the controller, preserving the original data types

## Links

[JIRA](https://codedotorg.atlassian.net/browse/P20-1012)

## Testing story

Tested locally. Example output pre-change. Note the first event, where the `previous_owner_id` is `203` and the `new_owner_id` is `"203"`:
```
Logging Event: {"user_id":203,"custom_ids":{"user_type":"teacher"},"event_name":"lti_section_owner_changed","event_value":"lti_section_owner_changed","metadata":{"section_id":78,"previous_owner_id":203,"new_owner_id":"203","lms_name":"canvas_cloud"}}
Logging Event: {"user_id":203,"custom_ids":{"user_type":"teacher"},"event_name":"lti_section_owner_changed","event_value":"lti_section_owner_changed","metadata":{"section_id":79,"previous_owner_id":203,"new_owner_id":"209","lms_name":"canvas_cloud"}}
```

Example post-change. No event is published for the section that didn't have an actual change in ownership, and the event for the changed section does have numbers in all the number fields.
```
Logging Event: {"user_id":203,"custom_ids":{"user_type":"teacher"},"event_name":"lti_section_owner_changed","event_value":"lti_section_owner_changed","metadata":{"section_id":81,"previous_owner_id":203,"new_owner_id":209,"lms_name":"canvas_cloud"}}
```

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb [hunk 1]
```diff
@@ -35,14 +35,14 @@ class Lti::V1::SectionsControllerTest < ActionController::TestCase
...
-    patch :bulk_update_owners, params: {section_owners: {lti_section.id => new_owner.id}}, format: :json
+    patch :bulk_update_owners, params: {section_owners: {lti_section.id => new_owner.id}.to_json}, format: :json
     assert_response :ok
     refute section.reload.user_id == @user.id
     assert section.instructors.include? @user
   end
 
   test 'returns a 404 if a section is not found' do
-    patch :bulk_update_owners, params: {section_owners: {'foo' => @user.id}}, format: :json
+    patch :bulk_update_owners, params: {section_owners: {'foo' => @user.id}.to_json}, format: :json
```
Comment (wilkie) [inline dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb:83]:
```suggestion
    Metrics::Events.expects(:log_event).never
    patch :bulk_update_owners, params: {section_owners: {'foo' => @user.id}.to_json}, format: :json
```

File: dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb [hunk 2]
```diff
@@ -35,14 +35,14 @@ class Lti::V1::SectionsControllerTest < ActionController::TestCase
     lti_section = create :lti_section, section: section
     create :section_instructor, section: section, instructor: new_owner
 
-    patch :bulk_update_owners, params: {section_owners: {lti_section.id => new_owner.id}}, format: :json
+    patch :bulk_update_owners, params: {section_owners: {lti_section.id => new_owner.id}.to_json}, format: :json
```
Comment (wilkie) [inline dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb:75]:
```suggestion
    Metrics::Events.expects(:log_event).with(
      has_entries(
        user: @user,
        event_name: 'lti_section_owner_changed',
        metadata: {
          section_id: section.id,
          previous_owner_id: @user.id,
          new_owner_id: new_owner.id,
          lms_name: lti_section.lti_course.lti_integration.platform_name,
        },
      )
    )
    patch :bulk_update_owners, params: {section_owners: {lti_section.id => new_owner.id}.to_json}, format: :json
```

File: dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb [hunk 3]
```diff
@@ -24,7 +24,7 @@ class Lti::V1::SectionsControllerTest < ActionController::TestCase
     other_teacher = create :teacher
     section = create :section, user: other_teacher
     lti_section = create :lti_section, section: section
-    patch :bulk_update_owners, params: {section_owners: {lti_section.id => @user.id}}, format: :json
+    patch :bulk_update_owners, params: {section_owners: {lti_section.id => @user.id}.to_json}, format: :json
```
Comment (wilkie) [inline dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb:52]:
```suggestion
    Metrics::Events.expects(:log_event).never
    patch :bulk_update_owners, params: {section_owners: {lti_section.id => @user.id}.to_json}, format: :json
```

File: dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb [hunk 4]
```diff
@@ -14,7 +14,7 @@ class Lti::V1::SectionsControllerTest < ActionController::TestCase
     create :section_instructor, section: section_one, instructor: new_owner
     create :section_instructor, section: section_two, instructor: new_owner
 
-    patch :bulk_update_owners, params: {section_owners: {lti_section_one.id => new_owner.id, lti_section_two.id => new_owner.id}}, format: :json
+    patch :bulk_update_owners, params: {section_owners: {lti_section_one.id => new_owner.id, lti_section_two.id => new_owner.id}.to_json}, format: :json
```
Comment (wilkie) [inline dashboard/test/controllers/lti/v1/lti_sections_controller_test.rb:41]:
```suggestion
    Metrics::Events.expects(:log_event).with(
      has_entries(
        user: @user,
        event_name: 'lti_section_owner_changed',
        metadata: {
          section_id: section_one.id,
          previous_owner_id: @user.id,
          new_owner_id: new_owner.id,
          lms_name: lti_section_one.lti_course.lti_integration.platform_name,
        },
      )
    )
    Metrics::Events.expects(:log_event).with(
      has_entries(
        user: @user,
        event_name: 'lti_section_owner_changed',
        metadata: {
          section_id: section_two.id,
          previous_owner_id: @user.id,
          new_owner_id: new_owner.id,
          lms_name: lti_section_two.lti_course.lti_integration.platform_name,
        },
      )
    )
    patch :bulk_update_owners, params: {section_owners: {lti_section_one.id => new_owner.id, lti_section_two.id => new_owner.id}.to_json}, format: :json
```

**Review by wilkie — COMMENTED** [review]
It looks good! I left some suggested edits to add testing that your event is logged (or not logged) in your scenarios so we can cover the thing you are fixing.

**Comment by carl-codeorg** [comment]
> It looks good! I left some suggested edits to add testing that your event is logged (or not logged) in your scenarios so we can cover the thing you are fixing.

Great suggestion, thank you @wilkie!

**Review by nicklathe — APPROVED** [review]

**Pull request merged**